### PR TITLE
Isdb 548 case reference screen

### PR DESCRIPTION
**Pull request by hmcts-abduldaim**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by yogendraU — CHANGES_REQUESTED** [review]

File: src/main/app/case/C100CaseApi.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export interface DocumentUploadResponse {
+  status: string;
+  document: {
+    document_url: string;
+    document_binary_url: string;
+    document_filename: string;
+    document_hash: string;
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/app/case/C100CaseApi.ts:7]: It wasn't added by me. it was already added prior to time when I have received the ticket.

File: src/main/app/controller/AppRequest.ts [hunk 2]
```diff
@@ -8,6 +10,7 @@ import { FormError } from '../form/Form';
 export interface AppRequest<T = Partial<Case>> extends Request {
   session: AppSession;
   locals: {
+    C100Api: any;
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/app/controller/AppRequest.ts:13]: It wasn't added by me. it was already added prior to time when I have received the ticket.

File: src/main/app/controller/GetController.ts [hunk 3]
```diff
@@ -45,16 +47,26 @@ export class GetController {
       },
     });
 
-    if (req.session?.errors) {
-      req.session.errors = undefined;
+    if (req.originalUrl === UPLOAD_DOCUMENT || req.originalUrl === DATA_VERIFICATION) {
+      console.log({ msg: 'triggering data disclosure as none' });
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/app/controller/GetController.ts:51]: Used successfully

File: src/main/app/form/validation.ts [hunk 4]
```diff
@@ -0,0 +1,246 @@
+import dayjs from 'dayjs';
+import customParseFormat from 'dayjs/plugin/customParseFormat';
+import { validate as isValidEmail } from 'email-validator';
+
+import { Case, CaseDate } from '../case/case';
+import { AllowedFileExtentionList, C100MaxFileSize, OtherName } from '../case/definition';
+
+dayjs.extend(customParseFormat);
+
+export type Validator = (
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/app/form/validation.ts:10]: cleaned

File: src/main/app/s2s/rpeAuth.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  private static rpeAdaptor = (): AxiosInstance => {
+    return axios.create({
+      baseURL: RpeApi._BASEURL,
+    });
+  };
+
+  public static async getRpeToken() {
+    try {
+      const Response: AxiosResponse = await RpeApi.rpeAdaptor().post('/testing-support/lease', {
+        microservice: 'fis_ds_update_web',
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/app/s2s/rpeAuth.ts:17]: resolved

File: src/main/modules/fileupload/index.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { Application } from 'express';
+import fileUpload from 'express-fileupload';
+
+/**
+ * @FileUpload
+ * sets up the fileUpload
+ */
+export class FileUpload {
+  public enableFor(app: Application): void {
+    // const DefaultFileUploadSize = config.get<number>('documentManagement.uploadSizeinMB');
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/modules/fileupload/index.ts:10]: resolved

File: src/main/steps/edgecase/casefinder/content.test.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  subtitle: 'Case reference number - welsh',
+  caseNameHint: 'This number will be 16 digits long - welsh',
+  errorSummaryMessage: 'There is a problem',
+  errors: {
+    applicantCaseId: {
+      required: 'There is a problem. Please enter a case reference number - welsh',
+    },
+    caseNotFound: {
+      required:
+        "Some of the information yo have given doesn't match our records. Please enter the right value and retry - welsh",
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/steps/edgecase/casefinder/content.test.ts:31]: resolved

File: src/main/steps/edgecase/casefinder/content.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  subtitle: 'Case reference number',
+  caseNameHint: 'This number will be 16 digits long',
+  errorSummaryMessage: 'There is a problem',
+  errors: {
+    applicantCaseId: {
+      required: 'There is a problem. Please enter a case reference number',
+    },
+    caseNotFound: {
+      required:
+        "Some of the information yo have given doesn't match our records. Please enter the right value and retry",
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/steps/edgecase/casefinder/content.ts:18]: resolved

File: src/main/steps/edgecase/casefinder/postController.ts [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+PostDocumentUploader method */
+@autobind
+export default class UploadDocumentController extends PostController<AnyObject> {
+  constructor(protected readonly fields: FormFields | FormFieldsFn) {
+    super(fields);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  public async serverCallForCaseIdValidations(req: AppRequest<AnyObject>) {
+    const baseURL = `http://localhost:3100/case/dss-orchestration/${req.body.applicantCaseId}`;
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/steps/edgecase/casefinder/postController.ts:23]: the main api will pulled for config.

File: src/main/steps/edgecase/casefinder/template.njk [hunk 10]
```diff
@@ -0,0 +1,56 @@
+{% extends "common/template.njk" %}
+{% from "govuk/components/error-summary/macro.njk" import govukErrorSummary %}
+{% block page_content %}
+
+{% if sessionError.length > 0 %}
+{% set errorList = []%}
+{% set formError = ''%}
+{% set formErrorMessage = ''%}
+{% set formInputError = ''%}
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/steps/edgecase/casefinder/template.njk:9]: I guess we discussed the solution before implementing it

File: src/main/steps/edgecase/citizen-data-verification/content.ts [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+            name: 'year',
+            value: year,
+            classes: 'govuk-input--width-4',
+            attributes: { maxLength: 4, pattern: '[0-9]*', inputMode: 'numeric' },
+          },
+        ],
+        validator: (value, formData) => (formData['DOB'] === '' ? isDateInputInvalid(value as CaseDate) : ''),
+      },
+    };
+  } else {
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/steps/edgecase/citizen-data-verification/content.ts:100]: they both are different one is taking argument

File: src/main/app/fileUpload/documentManager.ts [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const uploadDocument = async (formData, s2sToken) => {
+  const baseURL = config.get('api.cos') + DOCUMENT_MANAGEMENT_CONFIGURATIONS.UPLOAD_URL;
+  const response = await axios.post(baseURL, formData, {
+    headers: {
+      ServiceAuthorization: `Bearer ${s2sToken}`,
+      ...formData.getHeaders(),
+    },
+    httpsAgent: new https.Agent({ rejectUnauthorized: false }),
```
[the remark this answers is not in the document]
Comment (hmcts-abduldaim) [reply src/main/app/fileUpload/documentManager.ts:19]: ok

**Review by yogendraU — APPROVED** [review]

**Review by rahulsanjava19 — APPROVED** [review]

**Pull request merged**